protocol/contracts/nexus/Relic.sol

```diff
@@ -0,0 +1,344 @@
+        require(whitelisted[msg.sender], "You cannot own a Relic yet");
+        uint256 tokenId = _tokenIdCounter.current();
+        enclaves[tokenId] = _selectedEnclave;
+        _tokenIdCounter.increment();
+        _safeMint(msg.sender, tokenId);
+
+        whitelisted[msg.sender] = false;
+
+        emit MintRelic(msg.sender, uint256(_selectedEnclave));
+    }
+
+    // Allows users to equip Shards into their Relics
+    function batchEquipShard(
+        uint256 _targetRelic,
+        uint256[] memory _itemIds,
+        uint256[] memory _amounts
+    ) external nonReentrant {
+        require(
+            msg.sender == ownerOf(_targetRelic),
+            "You don't own this Relic"
+        );
+
+        SHARDS.equipShard(msg.sender, _itemIds, _amounts);
+
+        for (uint256 i = 0; i < _itemIds.length; i++) {
+            balances[_targetRelic][_itemIds[i]] += _amounts[i];
+        }
+    }
+    // Allows users to unequip Shards from their Relic
+    function batchUnequipShard(
+        uint256 _targetRelic,
+        uint256[] memory _itemIds,
+        uint256[] memory _amounts
+    ) external nonReentrant {
+        require(
+            msg.sender == ownerOf(_targetRelic),
+            "You don't own this Relic"
+        );
+
+        SHARDS.unEquipShard(msg.sender, _itemIds, _amounts);
+
+        for (uint256 i = 0; i < _itemIds.length; i++) {
+            balances[_targetRelic][_itemIds[i]] -= _amounts[i];
+        }
+    }
+
+    // allows whitelisted contract to mint to an address
+    function mintFromContract(address _to, Enclave _selectedEnclave) external {
+        require(
+            whitelistedContracts[msg.sender],
+            "This contract is not authorised to mint"
+        );
+        uint256 tokenId = _tokenIdCounter.current();
+        enclaves[tokenId] = _selectedEnclave;
+        _tokenIdCounter.increment();
+        _safeMint(_to, tokenId);
+    }
+
+    // external temple contract whitelisting for Relic minting
+    function whitelistTemplar(address _toWhitelist) external {
+        require(msg.sender == whitelisterAddress, "Not authorised");
+        whitelisted[_toWhitelist] = true;
+    }
+    // external temple contracts giving out XP points to your Relic
+    function givePoints(uint256 _amount, uint256 _relicId) external {
+        require(whitelistedContracts[msg.sender], "Not authorised");
+        relicXP[_relicId] += _amount;
+        emit GivePoints(_relicId, _amount);
+    }
+
+    //------- Public -------//
+
+    function supportsInterface(bytes4 interfaceId)
+        public
+        view
+        override(IERC165, ERC721, ERC721Enumerable)
+        returns (bool)
+    {
+        return super.supportsInterface(interfaceId);
+    }
+
+    function onERC1155Received(
+        address,
+        address,
+        uint256,
+        uint256,
+        bytes memory
+    ) public virtual override returns (bytes4) {
+        return this.onERC1155Received.selector;
```

Comment:
Should this guard to make sure the only ERC1155 it can accept is the Shards contract?